### PR TITLE
Multi-stage-build

### PR DESCRIPTION
**Pull request by kvinayakabaskaran**
## Summary

Brief explanation of the proposed changes.

## Basic example

Include a basic example, screenshots, or links.

## Motivation

Why are we doing this? What use cases does it support? What is the expected outcome?

## Checks

- [x] Read [Create a Pull Request](https://getdoks.org/docs/contributing/how-to-contribute/#create-a-pull-request)
- [x] Supports all screen sizes (if relevant)
- [x] Supports both light and dark mode (if relevant)
- [x] Passes `npm run test`

### CODE REVIEW

File: content/en/blog/multi-stage-docker-build/index.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+---
+title: "Multi Stage Docker Build 🐳 "
+description: "Hsi"
+excerpt: "Optimize Docker images with multi-stage builds to reduce size,"
+date: 2024-09-26T21:22:41+00:00 
+lastmod: 2024-09-26T21:22:41+00:00 
+draft: false
+weight: 90
+images: [multi-stage-image.jpg]
+categories: ["News"]
```
Comment (satrox28) [inline content/en/blog/multi-stage-docker-build/index.md]: Can you change the categories to the one related to the article.

File: content/en/blog/multi-stage-docker-build/index.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+---
+title: "Multi Stage Docker Build 🐳 "
+description: "Hsi"
+excerpt: "Optimize Docker images with multi-stage builds to reduce size,"
+date: 2024-09-26T21:22:41+00:00 
+lastmod: 2024-09-26T21:22:41+00:00
```
Comment (satrox28) [inline content/en/blog/multi-stage-docker-build/index.md]: Can you change all timestamp.

File: content/en/blog/multi-stage-docker-build/index.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
+---
+title: "Multi Stage Docker Build 🐳 "
+description: "Hsi"
```
Comment (satrox28) [inline content/en/blog/multi-stage-docker-build/index.md]: Description check

File: content/en/blog/multi-stage-docker-build/index.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+FROM node:22 AS build
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy package.json and package-lock.json to the working directory
+COPY package.json ./
+
+# Install dependencies using npm
+RUN npm i
```
Comment (satrox28) [inline content/en/blog/multi-stage-docker-build/index.md]: Check npm command

**Comment by anishbista60** [comment]
@kvinayakabaskaran please rebase these multiple commit to single one .

File: content/en/blog/multi-stage-docker-build/index.md (hunk 3 above)
Comment (kvinayakabaskaran) [reply content/en/blog/multi-stage-docker-build/index.md]: I have update the description

File: content/en/blog/multi-stage-docker-build/index.md (hunk 1 above)
Comment (kvinayakabaskaran) [reply content/en/blog/multi-stage-docker-build/index.md]: I have updated the categories

File: content/en/blog/multi-stage-docker-build/index.md (hunk 4 above)
Comment (kvinayakabaskaran) [reply content/en/blog/multi-stage-docker-build/index.md]: I have updated the code

**Comment by kvinayakabaskaran** [comment]
I have rebased the multiple commits to single and update the content

File: content/en/blog/multi-stage-docker-build/index.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
+---
+title: "Multi Stage Docker Build 🐳 "
+description: "Multi-stage Docker builds allow you to optimize image size by using multiple stages in a single Dockerfile, copying only the necessary files from each stage."
+excerpt: "Optimize Docker images with multi-stage builds to reduce size,"
+date: 2024-10-22T15:51:05
+lastmod: 2024-10-22T15:51:05 
+draft: false
+weight: 90
```
Comment (anishbista60) [inline content/en/blog/multi-stage-docker-build/index.md]: change it to 100.

**Comment by anishbista60** [comment]
@kvinayakabaskaran squash the commits into single one and And please rebase your branch with main because this branch is missing one of the fix related to dark theme which is already merged in main.  Thank you.

File: content/en/blog/multi-stage-docker-build/index.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+# Stage 2: Production-ready stage
+FROM node:22-slim
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy built artifacts from the previous stage
+COPY - from=build /app/dist ./dist
+COPY - from=build /app/node_modules ./node_modules
+COPY - from=build /app/package.json ./package.json
```
Comment (anishbista60) [inline content/en/blog/multi-stage-docker-build/index.md]:
```suggestion
COPY --from=build /app/dist ./dist
COPY --from=build /app/node_modules ./node_modules
COPY --from=build /app/package.json ./package.json
```

**Comment by anishbista60** [comment]
please address everything then will merge it. Thank you

**Comment by anishbista60** [comment]
please change the weight to 100 because we are following the standard and rebase it main because it is missing changes related to the dark theme.

**Comment by kvinayakabaskaran** [comment]
> rebase it main because it is missing changes related to the dark theme.

sure, i will do it

**Review by satrox28 — APPROVED** [review]

**Pull request merged**